realm/realm-library/src/androidTest/java/io/realm/RealmTest.java

```diff
@@ -2696,4 +2696,54 @@ public void execute(Realm realm) {
             handlerThread.quit();
         }
     }
+
+    public void testHandlerControllerThrowNullPointExceptionOnNonLooperThread() {
+        final CountDownLatch signalTestFinished = new CountDownLatch(1);
+        Thread thread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                Realm realm = Realm.getInstance(testConfig);
+                try {
+                    realm.handlerController.removeAllChangeListeners();
+                    fail("");
+                } catch (NullPointerException ignored) {
```

Comment:
Uh. my bad. This case can be simpler by just checking if `realm.handlerController` is null.